lib/avrora/storage/registry.ex

```diff
@@ -93,6 +94,35 @@ defmodule Avrora.Storage.Registry do
   @spec configured?() :: true | false
   def configured?, do: !is_nil(registry_url())
 
+  defp extract_references(response) do
+    case Map.fetch(response, "references") do
+      {:ok, references} ->
+        references_map =
+          Enum.reduce(references, %{}, fn cr, map ->
+            get_reference_schema(map, cr["subject"])
+          end)
+
+        {:ok, references_map}
+
+      :error ->
+        {:ok, :schema_without_references}
+    end
+  end
+
+  defp get_reference_schema(map, subject) do
+    case get(subject) do
+      {:ok, schema_map} -> Map.put(map, schema_map.full_name, schema_map.json)
+      {:error, error} -> throw(error)
+    end
+  end
+
+  defp make_references_lookup_fun(references) do
+    case references do
+      :schema_without_references -> &Avrora.Schema.reference_lookup/1
+      _ -> &Map.fetch(references, &1)
+    end
+  end
```

Comment:
Not sure why the `credo` silent about it, but for such a case `if/else` should be used

```elixir
    if references == :no_references,
      do: &Avrora.Schema.reference_lookup/1
      else: &Map.fetch(references, &1)
```

And then we can return `{:ok, %{}}` in `extract_references` and use the same `if/else`

```elixir
     if references == %{},
       do: &Avrora.Schema.reference_lookup/1
       else: &Map.fetch(references, &1)
    end
```

and from here we can go to pattern matching

```suggestion
  def make_reference_lookup_function(%{}), do: &Avrora.Schema.reference_lookup/1
  def make_reference_lookup_function(references), do: &Map.fetch(references, &1)
```